### PR TITLE
Changes the missing food icon test to cover ALL /obj's

### LINKED ISSUE #71953: Secure breifcase suddenly invis
## Reproduction:
I was messing with the breifcase's code then suddenly it turned invisible?
image

### LINKED ISSUE #71983: Mothic ingredient boxes (And apparently anything that comes in a box from the produce console) are invisible
Reporting client version: 514.1589

## Round ID:
[196232](https://scrubby.melonmesa.com/round/196232)

## Testmerges:
- [Partially reverts moving up/down whilst in a movable from #70431](https://www.github.com/tgstation/tgstation/pull/71527)
- [Avoid wasted work creating body parts that are just going to be destroyed [0.3s of init]](https://www.github.com/tgstation/tgstation/pull/71916)

## Reproduction:
Another storage item missing its sprite, this time it was the mothic supply boxes from the produce console. The Tirizan ones supposedly have the same problem

### PR DESCRIPTION
**Pull request by ShizCalev**
Might as well cover everyyyyyyyyything. :)

Fixes https://github.com/tgstation/tgstation/issues/71953
Fixes https://github.com/tgstation/tgstation/issues/71983

:cl: ShizCalev
code: We now unit test all /obj's for missing icons. :)
/:cl:

Closes https://github.com/tgstation/tgstation/projects/17#card-86939061 / unit tests for #71469

### CODE REVIEW

**Comment by dragomagol** [comment]
You're the best

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 1]
```diff
@@ -292,6 +292,7 @@
 	projectile_type = /obj/projectile/tentacle
 	caliber = CALIBER_TENTACLE
 	icon_state = "tentacle_end"
+	icon = 'icons/obj/weapons/guns/projectiles.dmi'
```
Comment (ShizCalev) [inline code/modules/antagonists/changeling/powers/mutations.dm:295]: have a feeling this one probbbbbbably doesn't actually need an icon, will check later

**Review by MrMelbert — COMMENTED** [review]
Didn't peek at the sprites themselves but everything else (mostly) seemed good.

File: code/modules/unit_tests/missing_icons.dm [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		if(icon_exists(icon, icon_state))
+			continue
+
+		bad_list[icon] += list(icon_state)
+
+		var/match_message = (icon_state in possible_icon_states) ? TRUE : null
+		if(match_message)
+			match_message = null
+			for(var/file_place in possible_icon_states[icon_state])
+				match_message += (match_message ? " & '[file_place]'" : " - Matching sprite found in: '[file_place]'")
```
[the remark this answers is not in the document]
Comment (ShizCalev) [reply code/modules/unit_tests/missing_icons.dm]: shhhhh

**Review by MrMelbert — APPROVED** [review]
Code/test is good. Some edge cases are uncovered but that's fine for now, it's a wide test. 

If someone else goes over the DMIs it's fine to merge

I'll nitpick this though
